lib/livebook/zta/teleport.ex

```diff
@@ -0,0 +1,87 @@
+defmodule Livebook.ZTA.Teleport do
+  use GenServer
+  require Logger
+
+  defstruct [:name, :req_options, :jwks]
+
+  @renew_afer 24 * 60 * 60 * 1000
+  @fields %{"sub" => :id, "username" => :username}
+  @assertion "teleport-jwt-assertion"
+  @well_known_jwks_path "/.well-known/jwks.json"
+
+  def start_link(opts) do
+    url = String.trim(opts[:identity][:key], "/")
+    full_path = "#{url}#{@well_known_jwks_path}"
+    options = [req_options: [url: full_path]]
+    GenServer.start_link(__MODULE__, options, name: opts[:name])
+  end
+
+  @impl true
+  def init(options) do
+    state = struct!(__MODULE__, options)
+
+    {:ok, %{state | jwks: renew_jwks(state.req_options)}}
```

Comment:
So if I understood correctly, we want to fetch the `jwks` on init and let it crash if there is something wrong with the jwks or the endpoint?

Reply:
Follow the template on the google_iap.ex one. I think that will be the best reference for this one. :)

Reply:
Alright, so in [Livebook.ZTA.GoogleIAP.init/1](https://github.com/livebook-dev/livebook/blob/626ea9781a99a5d0c1f25f5ebbccdb095597875c/lib/livebook/zta/google_iap.ex#L27) we call [keys/1](https://github.com/livebook-dev/livebook/blob/626ea9781a99a5d0c1f25f5ebbccdb095597875c/lib/livebook/zta/google_iap.ex#L40-L45) which makes HTTP request with bang.

I followed the same style here